torch_xla/csrc/ops/custom_sharding.h

```diff
@@ -7,14 +7,24 @@ namespace torch_xla {
 
 class CustomSharding : public XlaNode {
  public:
+  enum class Type {
+    kSharding,
+    kSPMDFullToShardShape,
+    kSPMDShardToFullShape,
+  };
```

Comment:
This enum is really confusing, can you add some comment around what they actually does? I was reading the SPMD code again, this op itself only means we want to shard the underlying value and the actual sharding resides in the XlaTensor or Based XLAIR object?

Reply:
Right, this is just the name of the custom call. The sharding annotation is in XlaTensor as normal. I can add more explanations.

Reply:
Maybe we can annotate explicilty that this is sharding type for custom call in the enum class name or somethinhg.

Reply:
I guess the current approach sort of does it already? Can you be more specific? @yeounoh

Reply:
I agree, `Type` is already defined under `CustomSharding`